### PR TITLE
Networking

### PR DESCRIPTION
**Pull request by DSVAG**
Add retrofit, replaced glide with coil

### CODE REVIEW

File: app/src/main/java/com/dsvag/androidacademyproject/ui/moviedetails/CastAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            itemBinding.name.text = cast.name
+
+            val url = "https://image.tmdb.org/t/p/h632" + cast.profilePath
+
+            itemBinding.photo.load(url) {
+                crossfade(true)
+                error(R.drawable.ic_launcher_foreground)
+            }
+
+            itemBinding.root.setOnClickListener {
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/ui/moviedetails/CastAdapter.kt:56]: setOnClickListener here will create an instance of the listener on each call onBindViewHolder. If moving it to onCreateViewHolder - will be created one listener on each ViewHolder instance.

File: app/src/main/java/com/dsvag/androidacademyproject/ui/movies/MovieAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            itemBinding.rating.rating = result.voteAverage.toFloat() / 2
+
+            val url = "https://image.tmdb.org/t/p/w780" + result.posterPath
+
+            itemBinding.preview.load(url) {
+                crossfade(true)
+                error(R.drawable.ic_launcher_foreground)
+            }
+
+            itemBinding.root.setOnClickListener {
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/ui/movies/MovieAdapter.kt:60]: Many unnecessary allocations on scrolling (above described).

File: app/src/main/java/com/dsvag/androidacademyproject/ui/movies/MoviesViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+                if (response.isSuccessful && response.body() != null) {
+                    val value = _mutableResult.value?.plus(response.body()!!.results)
+
+                    maxPageCounter = response.body()!!.totalResults
+                    _mutableResult.postValue(value)
+                    setState(State.Success)
+                } else {
+                    setState(State.Error(response.errorBody().toString()))
+                }
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/ui/movies/MoviesViewModel.kt:57]: It seems that fetch functions can be better (boilerplate, nullable values).

**Review by pcfaktor — APPROVED** [review]
1. Search does not work now?

Cool. Approval.

File: app/src/main/java/com/dsvag/androidacademyproject/data/repositories/MovieRepository.kt [hunk 4]
```diff
@@ -1,14 +1,41 @@
...
+import retrofit2.Response
 import javax.inject.Inject
 
 class MovieRepository @Inject constructor(
-    @ApplicationContext private val context: Context,
+    private val apiMovieService: ApiMovieService
 ) {
-    suspend fun fetchMovies() = withContext(Dispatchers.IO) { loadMovies(context) }
+    suspend fun getNowPlaying(page: Int): Response<Request> {
+        return apiMovieService.getNowPlaying(API_KEY, page)
```
[the remark this answers is not in the document]
Comment (DSVAG) [reply app/src/main/java/com/dsvag/androidacademyproject/data/repositories/MovieRepository.kt:15]: Need to write **_CallAdapter_** and implement it _**CallAdapter.Factory**_?

**Pull request merged**